### PR TITLE
Dashboard loader supports transformations & unit tests

### PR DESCRIPTION
**Pull request by ivanortegaalba**
**Changes:**
* Apply panel transformations to query response data
* Refactor `DashboardLoader` to extract the translators out of the main class
* Add unit tests to `DashboardLoader()`

### CODE REVIEW

File: public/app/features/scenes/dashboard/DashboardsLoader.test.ts [hunk 1]
```diff
@@ -1,13 +1,352 @@
...
+      const rowScene = scene.state.body.state.children[0] as SceneGridRow;
+      expect(rowScene).toBeInstanceOf(SceneGridRow);
+      expect(rowScene.state.title).toEqual(row.title);
+      expect(rowScene.state.placement?.y).toEqual(row.gridPos!.y);
+      expect(rowScene.state.isCollapsed).toEqual(row.collapsed);
+      expect(rowScene.state.children).toHaveLength(1);
+      expect(rowScene.state.children[0]).toBeInstanceOf(VizPanel);
+    });
+
+    it('should create panels within expanded row', () => {
```
Comment (torkelo) [inline public/app/features/scenes/dashboard/DashboardsLoader.test.ts:185]: The setup (state) for the panels is pretty long with lots of things that are not relevant for the test. think a function that takes a partial Panel would help here, so we can only specify the gridPos

**Review by torkelo — APPROVED** [review]
This looks great! Only nitpick is to extract a function to create panel models from a partial, to reduce the setup code a bit.

File: public/app/features/scenes/dashboard/DashboardsLoader.test.ts (hunk 1 above)
Comment (dprokop) [reply public/app/features/scenes/dashboard/DashboardsLoader.test.ts:185]: Agree, we have a helper for tat that should be useful: [createPanelJSONFixture](https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/state/__fixtures__/dashboardFixtures.ts#L32)

**Review by dprokop — APPROVED** [review]
Looks great, nice tests!

**Pull request merged**